### ISSUE TITLE
alr update fails: Dependency resolution failed

### ISSUE THREAD

**Issue by stcarrez** [alire-project/alire#377]
Yet another useful improvement for Alire. Sometimes you get the following error when you run the `alr update` command:

```
-> Dependency resolution failed
ERROR: Update failed
ERROR: alr update unsuccessful
```

The `-v` option does not help. The `-vv` option requires deep understanding of how Alire works.

It would be nice to have an explanation about the dependency failure.

Of course, I made a mistake, but which crate is having a wrong dependency and for which other crate? That's is the question!

**Comment by mosteo** [comment]
Yes, I think this is a duplicate of #220.

**Comment by Fabien-Chouteau** [comment]
There are indeed many places where `alr` output can be more user-friendly.
Thank you for reporting those :+1: 

@mosteo maybe we should have a specific GitHub issue label for this kind of problem so that we can list them when we have time to work on usability.

**Comment by mosteo** [comment]
:+1: Created the `usability` label for this

**Comment by mosteo** [comment]
Implemented by #451. Now you get a summary of the dependency solution, where any problematic dependency will also be highlighted as missing.
